### PR TITLE
Add troubleshooting guideline link to PNSE in Quic

### LINKED ISSUE #87970: Make libmsquic load failures actionable
When user tries to use HTTP/3 or Quic APIs and they fail to load libmsquic (esp. on Linux), we should provide steps to troubleshoot.
1. If the problem was insufficient libmsquic version, say so in exception text.
2. If the problem was general loading failure, give user link to our troubleshooting guide, which we can edit over time, incl. instructions which tools to use to troubleshoot.

It might require exposing new API from Quic APIs. Or we could use it as text of PlatformNotSupportedException we throw from there.
Customer of the API is HttpClient and also any other (external) Quic API user, incl. Kestrel.

### PR DESCRIPTION
**Pull request by liveans**
Add troubleshooting guideline to PlatformNotSupportedException in Quic
Fixes #87970

### CODE REVIEW

File: src/libraries/System.Net.Quic/src/System/Net/Quic/QuicConnection.cs [hunk 1]
```diff
@@ -60,7 +60,7 @@ public static ValueTask<QuicConnection> ConnectAsync(QuicClientConnectionOptions
     {
         if (!IsSupported)
         {
-            throw new PlatformNotSupportedException(SR.Format(SR.SystemNetQuic_PlatformNotSupported, MsQuicApi.NotSupportedReason));
+            throw new PlatformNotSupportedException(SR.Format(SR.SystemNetQuic_PlatformNotSupported, MsQuicApi.NotSupportedReason ?? "General loading failure."));
```
Comment (ManickaP) [inline src/libraries/System.Net.Quic/src/System/Net/Quic/QuicConnection.cs:63]: Why this change? Can `NotSupportedReason` be null here?

Comment (liveans) [reply src/libraries/System.Net.Quic/src/System/Net/Quic/QuicConnection.cs:63]: Given that it's nullable string, I thought that should be fine to provide something like `General loading failure`, instead of having it `emtpy`.

File: src/libraries/System.Net.Quic/src/Resources/Strings.resx [hunk 2]
```diff
@@ -136,7 +136,7 @@
     <value>Stream aborted by peer ({0}).</value>
   </data>
   <data name="SystemNetQuic_PlatformNotSupported" xml:space="preserve">
-    <value>System.Net.Quic is not supported on this platform: {0}</value>
+    <value>System.Net.Quic is not supported on this platform: {0} For more information see: https://learn.microsoft.com/dotnet/fundamentals/networking/quic/quic-troubleshooting</value>
```
[the remark this answers is not in the document]
Comment (liveans) [reply src/libraries/System.Net.Quic/src/Resources/Strings.resx:139]: I thought we were going to add something over time, but I'm totally fine with this link as well.

File: src/libraries/System.Net.Quic/src/System/Net/Quic/QuicConnection.cs (hunk 1 above)
Comment (ManickaP) [reply src/libraries/System.Net.Quic/src/System/Net/Quic/QuicConnection.cs:63]: But can it be empty? Are there cases when it's actually null?

Comment (liveans) [reply src/libraries/System.Net.Quic/src/System/Net/Quic/QuicConnection.cs:63]: Looks like it can, if I'm not missing something.

https://github.com/dotnet/runtime/blob/81441a1b1abc46309ce472539aac83af7f18cd39/src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicApi.cs#L136-L142

**Review by ManickaP — APPROVED** [review]
